certbot-nginx/certbot_nginx/_internal/configurator.py

```diff
@@ -912,7 +915,7 @@ def restart(self):
         :raises .errors.MisconfigurationError: If either the reload fails.
 
         """
-        nginx_restart(self.conf('ctl'), self.nginx_conf)
+        nginx_restart(self.conf('ctl'), self.nginx_conf, self.conf('sleep-seconds'))
```

Comment:
This will sleep for *any* reload, even if it's from cleanup. Perhaps that is not necessary.

What gives me pause is how the nginx master will behave when it receives a `SIGHUP` when it is already in the middle of reloading the configuration. As far as I can understand the nginx code, (at most) one reload will end up queued behind one that is already running.

If there are back-to-back nginx challenges and choose not to sleep after the cleanup, then we may end up in the situation where we actually have to wait for between 1.0-2.0x the duration of `--nginx-sleep-interval`, since the original reload (from the cleanup restart) has not finished yet. There also might be a filesystem race, depending whether nginx reads config files lazily or not during parsing.

Because of those concerns, I have implemented this in a conservative way (always sleep for any reload), but it comes at the cost of speed.